STYLE.md

```diff
@@ -0,0 +1,270 @@
+Attributes exported by your modules should follow the same descriptive format as variable names.
+
+## Comments
+
+Only comment what is necessary.
+
+Single line comments: `#` or `//`.
+
+Multi-line comments: `/*` followed by `*/`.
+
+When commenting use two "//" and a space in front of the comment.
+
+```
+// CREATE ELK IAM ROLE 
+...
+```
+
+## Naming Conventions
+
+### File Names
+
+Create a separate resource file for each type of AWS resource. Similar resources should be defined in the same file and named accordingly.
+
+```
+ami.tf
+autoscaling_group.tf
+cloudwatch.tf
+iam.tf
+launch_configuration.tf
+providers.tf
+s3.tf
+security_groups.tf
+sns.tf
+sqs.tf
+user_data.sh
+variables.tf
+```
+
+### Parameter, Meta-parameter and Variable Naming
+
+ Only use an underscore (`_`) when naming Terraform resources like TYPE/NAME parameters and variables.
+
+ ```
+resource "aws_security_group" "security_group" {
+...
+ ```
+
+Variable names should be descriptive. Avoid abbreviations when reasonable.  Verbosity is a good thing.
+
+```hcl
+// bad
+variable "pub_sub_id" {}
+
+// good
+variable "public_subnet_id" {
+  // definition
+}
+```
+
+Boolean variables should follow the idiomatic pattern `is_<condition>`
+
+```hcl
+// bad
+variable "public" {}
+
+// bad
+variable "is_public" {
+  description = "Is the subnet public?"
+  type        = "string"
+  default     = false
+}
+```
+
+Terraform resources usually have an ID or some other attribute associated with them.  Variables for modules should refer to these attributes and match the actual name as much as possible.
+
+```hcl
+// bad
+variable "vpc_cidr" {}
+// -- or --
+variable "cidr_block" {}
+// -- or --
+variable "address_range" {}
+
+
+// good
+variable "vpc_cidr_block" {
+  // the vpc module exports an attribute named "cidr_block"
+  description = "The CIDR block associated with the VPC"
+  type = "string"
+}
+```
+
+If the variable has a particular unit of measure, feel free to add that in for additional clarity.
+
+```hcl
+// bad
+variable "root_disk_size" {}
+
+// good
+variable "root_disk_size_in_gb" {
+  // definition
+}
+```
+
+Variable names for lists should somehow indicate that they are a group of things.  You know... plural.
+
+```hcl
+// bad
+variable "public_subnet_id" {}
+
+// good
+variable "public_subnet_ids" {
+  // definition
+}
+```
+
+Use variable names/descriptions from the official resource as much as possible.
+
+```hcl
+// good
+// todo example
+
+// bad
+// todo example
+```
+
+Not prefix all variable names with lambda_ when the this is mainly a lambda function.
+
+```hcl
+// good
+// todo example
+
+// bad
+// todo example
+```
+
+###Resource Naming
+
+__Only use a hyphen (`-`) when naming the component being created.__
+
+ ```
+resource "aws_security_group" "security_group" {
+  name = "${var.resource_name}-security-group"
+...
+ ```
+
+__A resource's NAME should be the same as the TYPE minus the provider.__
+
+```
+resource "aws_autoscaling_group" "autoscaling_group" {
```

Comment:
I'm conflicted about this. On the one hand it is an easy-to-follow instruction, but as soon as you have two identical resources the `_autoscaling_group` part of the name becomes redundant (and that it is a autoscaling group is already clear from the resource type).